controllers/awscluster_controller.go

```diff
@@ -204,6 +230,20 @@ func (r *AWSClusterReconciler) SetupWithManager(mgr ctrl.Manager, options contro
 		WithOptions(options).
 		For(&infrav1.AWSCluster{}).
 		WithEventFilter(pausedPredicates(r.Log)).
+		WithEventFilter(
+			predicate.Funcs{
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					if e.ObjectOld.GetObjectKind().GroupVersionKind().Kind != "AWSCluster" {
+						return true
+					}
+
+					oldCluster := e.ObjectOld.(*infrav1.AWSCluster)
+					newCluster := e.ObjectNew.(*infrav1.AWSCluster)
+
+					return !reflect.DeepEqual(oldCluster.Spec, newCluster.Spec)
```

Comment:
Similar concerns here as with AWSMachine around potentially blocking reconciliations that we were previously relying on.